gui/src/app/pyodide/pyodideWorker/pyodideWorker.ts

```diff
@@ -0,0 +1,246 @@
+import { PyodideInterface, loadPyodide } from "pyodide";
+import {
+  MessageFromPyodideWorker,
+  isMessageToPyodideWorker,
+  PyodideWorkerMode,
+  PydodideWorkerStatus,
+} from "./pyodideWorkerTypes";
+
+let pyodideWorkerMode: PyodideWorkerMode | undefined = undefined;
+
+let pyodide: PyodideInterface | null = null;
+const loadPyodideInstance = async () => {
+  if (pyodide === null) {
+    if (!pyodideWorkerMode) {
+      throw Error("pyodideWorkerMode is not defined");
+    }
+    const packages =
+      pyodideWorkerMode === "data.py"
+        ? ["numpy", "micropip"]
+        : pyodideWorkerMode === "analysis.py"
+          ? ["numpy", "matplotlib", "pandas"]
+          : [];
+    pyodide = await loadPyodide({
+      indexURL: "https://cdn.jsdelivr.net/pyodide/v0.26.1/full",
+      stdout: (x: string) => {
+        sendStdout(x);
+      },
+      stderr: (x: string) => {
+        sendStderr(x);
+      },
+      packages,
+    });
+    if (pyodideWorkerMode === "data.py") {
+      const micropip = pyodide.pyimport("micropip");
+      await micropip.install("stanio");
+    }
+    if (pyodideWorkerMode === "analysis.py") {
+      const sp_load_draws_py = await fetchScript('/sp_load_draws.py')
+      pyodide.FS.writeFile("sp_load_draws.py", sp_load_draws_py, {
+        encoding: "utf-8",
+      });
```

Comment:
Again with written to file vs included in preamble

Reply:
I'm thinking of possibly having other files being imported in, and it's better to have them be stand-alone and imported in the main script. Again, trying to keep the main script as simple as possible.

Reply:
Some files may need to import others.

Reply:
And the more preamble we have, the more names are contaminating the namespace...